### ISSUE TITLE
How to assign keywords, title in set_properties

### ISSUE THREAD

**Issue by hulksyed07** [Paxa/fast_excel#30]
I am trying to set the keyword property in the excel file as 'Confidential' by using the below method

workbook = FastExcel.open('./dummy.xlsx')
workbook.set_properties({ keywords: 'Confidential' })

It gives me error that "Argument is not a valid pointer"

After Going through the gem it looks like that it expects a DocProperties Object

**d = Libxlsxwriter::DocProperties.new

d[:keywords] = 'Confidential'

workbook.set_properties(d)**

When i try to assign d[:keywords] = 'Confidential' then it gives me error that "Cannot set string field"

Could you please let me know how to assign properties

**Comment by hulksyed07** [comment]
@Paxa 
Also I am not able to add custom_properties
2.4.0 :003 > workbook = FastExcel.open('Hello_world.xlsx', constant_memory: true)

2.4.0 :006 > workbook.set_custom_property_string('Sensitivity', 'Confidential')
 => :no_error

**Comment by Paxa** [comment]
I think this is related https://github.com/ffi/ffi/issues/10
Need to make some setter methods... will try to make it easier and documented later.

For now I was able to make it works like this:
```ruby
require_relative 'fast_excel'

class Libxlsxwriter::DocProperties < FFI::Struct
  def keywords=(val)
    pos = offset_of(:keywords)

    if val.is_a?(String)
      val = FFI::MemoryPointer.from_string(val)
    end

    if val.nil?
      self.pointer.put_pointer(pos, FFI::MemoryPointer::NULL)
    elsif val.is_a?(FFI::MemoryPointer)
      self.pointer.put_pointer(pos, val)
    else
      fail("keywords= requires an FFI::MemoryPointer or nil")
    end

    val
  end
end

workbook = FastExcel.open("example_workbook_properties.xlsx", constant_memory: true)

props = Libxlsxwriter::DocProperties.new
props.keywords = 'Confidential'

workbook.set_properties(props)

# print properties
FastExcel.print_ffi_obj(workbook[:properties])

workbook.close
```

**Comment by Laykou** [comment]
@Paxa Is there any way how to make this more simple using hashes?

**Comment by Paxa** [comment]
@Laykou Not yet, PRs are welcome :)

**Comment by Laykou** [comment]
Yeah, this is a bit low-level to me with the C-like programming, pointers and FFI :P
